src/Sarif/SarifExtensionMethods.cs

```diff
@@ -258,7 +260,25 @@ public static string GetFileName(this Uri uri)
 
         public static string GetFilePath(this Uri uri)
         {
-            return uri.IsAbsoluteUri ? uri.LocalPath : uri.OriginalString;
+            string originalString = uri.OriginalString;
+            string originalStringWithoutQuery = originalString.Split('?')[0];
+
+            // 'uri.LocalPath' is not supported for RelativeUri.
+            if (!uri.IsAbsoluteUri)
+            {
+                return originalStringWithoutQuery;
+            }
+
+            if (Regex.IsMatch(originalString, URLPercentEncodingPattern))
+            {
+                // If the 'originalString' contains URL percent-encoding characters,
+                // 'uri.LocalPath' will decode them thus changing the path unexpectedly.
+                return originalStringWithoutQuery;
+            }
+            else
```

Comment:
>else [](http://example.com/codeflow?start=12&length=4)

you don't need an else block here, the previous conditional always returns.

Reply:
Remove the else block.